pysqldb3/geopackage.py

```diff
@@ -456,26 +437,55 @@ def read_gpkg(self, dbo, table = None, gpkg_tbl = None, schema = None, port = 54
 
             table = table.lower()
 
-            if dbo.table_exists(schema = schema, table = table):
+            gpkg_tbl_names[table] = gpkg_tbl
 
-                print(f'Deleting existing table {schema}.{table}')
-                dbo.drop_table(schema=schema, table=table)
+        else:
+
+            ####### BULK UPLOAD ########
```

Comment:
It seems to me this is doing too many things, it should only import 1 table, if there are many tables that need importing it should go to a bulk read function, or if you want you can set the input table to be a list and loop through that. 
i think rather than have the bulk call this function with bulk = true, i would move the bulk logic to the bulk read function and have this call that

Reply:
OK I have changed it in my latest update